controllers/helmchartinstall_controller.go

```diff
@@ -58,12 +78,41 @@ func (r *HelmChartInstallReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 
 	r.Log.V(0).Info("Hello World!", "instanceName", req.NamespacedName, "instance", i)
 
+	// FIXME TODO we are conflating the setup and render phases here, because originally
+	// I was writing this controller as an experiment for a controller which only supported
+	// rendering.
+	//
+	// Here's what we want in the long run, for a controller which supports both:
+	// - Setup phase: reads configuration, and sets up event handlers. An event handler has:
+	//   * A target CRD GVK (specified by the stack configuration for the behavior)
+	//   * A watcher (the controller would create this, to watch CRDs of the desired type)
+	//   * A set of hooks (specified by the stack configuration for the behavior). Each hook
+	//     takes in a set of configuration and the triggering CRD, and outputs some K8s resources
+	// - Render phase: this is what happens when a hook is triggered, and the hook runs
+	//   * Inputs:
+	//     ^ A claim, which is the object which triggered the render
+	//     ^ A behavior, which has all of the configuration for what should happen now
+	//       that the render has been triggered
+	//   * Outputs:
+	//     ^ Changes to new or existing K8s resources
+	//     ^ The result of the render (probably as the status of the claim)
+	//       + The job which was run
+	//       + The result of the job
+	//       + The logs of the job (maybe; it might be too much)
+
+	// TODO remaining functionality for the render phase:
+	// - Surface the job result somehow (for usability)
+	// - Support for deletion, unless we assume that garbage collection will do it for us
 	r.render(ctx, i)
 
 	return ctrl.Result{}, nil
```

Comment:
Nitpick: Lines 106 could simply be `return ctrl.Result{}, r.render(ctx, i)`

You may also consider using https://godoc.org/github.com/pkg/errors or https://blog.golang.org/go1.13-errors to add context to your errors when you return them. The Go 1.13 stdlib method is probably the future, but we use `pkg/errors` in Go. I am having a hard time getting behind the stdlib method because I strongly prefer the API of `pkg/errors`.